ml_genn/model.py

```diff
@@ -274,6 +273,20 @@ def get_kernel_times(self):
             'synapse_dynamics_time': self.g_model.synapse_dynamics_time,
         }
 
+    def summary(self):
+        """Print a summary of this model"""
+
+        # layers should already be topologically sorted
+        print('===== Summary of {} ====='.format(self.name))
+
+        for l in self.layers:
+            print('\nname: {}  shape: {}  type: {}'.format(
+                l.name, l.shape, l.__class__.__name__))
+
+            if isinstance(l, Layer):
+                for s in l.upstream_synapses:
+                    print('connections: {}'.format(
```

Comment:
If I understand it right these are the incoming connections; maybe it would be better to disambiguate by printing out "incoming connections" instead of "connections:"?

Reply:
I'll change it.